pkg/openwsn/contrib/radio_hal.c

```diff
@@ -0,0 +1,316 @@
+{
+    /* OpenWSN `len` accounts for the FCS field which is set by default by
+       netdev, so remove from the actual packet `len` */
+    iolist_t pkt = {
+        .iol_base = (void *)packet,
+        .iol_len = (size_t)(len - IEEE802154_FCS_LEN),
+    };
+
+    if (ieee802154_radio_write(openwsn_radio.dev, &pkt) < 0) {
+        LOG_ERROR("[openwsn/radio]: couldn't load pkt\n");
+    }
+}
+
+void radio_txEnable(void)
+{
+    int ret = ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
+                                                     IEEE802154_TRX_STATE_TX_ON);
+
+    if (ret) {
+        LOG_ERROR("[openwsn/radio]: request_set_trx_state failed %s\n",
+                  __FUNCTION__);
+    }
+    else {
+        while (ieee802154_radio_confirm_set_trx_state(openwsn_radio.dev) ==
+               -EAGAIN) {}
+        debugpins_radio_set();
+        leds_radio_on();
+    }
+}
+
+void radio_txNow(void)
+{
+    int ret = ieee802154_radio_request_transmit(openwsn_radio.dev);
+
+    if (ret) {
+        LOG_ERROR("[openwsn/radio]: request_set_trx_state failed %s\n",
+                  __FUNCTION__);
+    }
+    else {
+        /* Trigger startFrame manually if no IEEE802154_CAP_IRQ_TX_START */
+        if (!ieee802154_radio_has_irq_tx_start(openwsn_radio.dev)) {
+            debugpins_isr_set();
+            _txrx_event_capture_time = sctimer_readCounter();
+            debugpins_isr_clr();
+            openwsn_radio.startFrame_cb(_txrx_event_capture_time);
+        }
+    }
+}
+
+void radio_rxEnable(void)
+{
+    int ret = ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
+                                                     IEEE802154_TRX_STATE_TRX_OFF);
+
+    if (ret) {
+        LOG_ERROR("[openwsn/radio]: request_set_trx_state failed %s\n",
+                  __FUNCTION__);
+    }
+    else {
+        while (ieee802154_radio_confirm_set_trx_state(openwsn_radio.dev) ==
+               -EAGAIN) {}
+        debugpins_radio_set();
+        leds_radio_on();
+    }
+}
+
+void radio_rxNow(void)
+{
+    int ret = ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
+                                                     IEEE802154_TRX_STATE_RX_ON);
+
+    if (ret) {
+        LOG_ERROR("[openwsn/radio]: request_set_trx_state failed %s\n",
+                  __FUNCTION__);
+    }
+    else {
+        while (ieee802154_radio_confirm_set_trx_state(openwsn_radio.dev) ==
+               -EAGAIN) {}
+    }
+}
+
+void radio_getReceivedFrame(uint8_t *bufRead,
+                            uint8_t *lenRead,
+                            uint8_t maxBufLen,
+                            int8_t *rssi,
+                            uint8_t *lqi,
+                            bool *crc)
+{
+    ieee802154_rx_info_t rx_info;
+    size_t size = ieee802154_radio_read(openwsn_radio.dev, bufRead,
+                                        maxBufLen, &rx_info);
+
+    /* FCS is skipped by the radio-hal in the returned length, but
```

Comment:
I guess this is missing in the Radio HAL API documentation

Reply:
resolved